src/main/java/org/reaktivity/nukleus/http2/internal/stream/Http2ServerFactory.java

```diff
@@ -2164,21 +2216,23 @@ else if (headersDecoder.connectionError != null)
             }
         }
 
-        private void onDecodeData(
+        private int onDecodeData(
             long traceId,
             long authorization,
-            Http2DataFW http2Data)
+            DirectBuffer payload)
         {
-            final int streamId = http2Data.streamId();
-            final int dataLength = http2Data.dataLength();
-            final boolean endRequest = http2Data.endStream();
+            int progress = 0;
+
+            final int streamId = dataStreamId;
```

Comment:
Please pass the `streamId` to `onDecodeData` instead of relying on `dataStreamId` decoder state.
Please also rename `dataStreamId` to `decodedStreamId`.